### PR TITLE
Update doc about creating a SecurityHandler

### LINKED ISSUE #7533: Documentation talks about extending final RoleSecurityHandler
Hello,

I just noticed today that the [Security documentation](https://github.com/sonata-project/SonataAdminBundle/blob/4.x/docs/reference/security.rst) says that "If you want to change the handler behavior, extend Sonata\AdminBundle\Security\Handler\RoleSecurityHandler, and override the isGranted method." However, this class is final in 4.x and marked final since sonata-project/admin-bundle 3.52 in 3.x. So it is impossible.

### PR DESCRIPTION
**Pull request by AntoineRoue**
Related to #7533

### CODE REVIEW

**Comment by AntoineRoue** [comment]
Looks like there is a problem with the workflow, the failed check cannot be due to commit.

**Comment by VincentLanglet** [comment]
> Looks like there is a problem with the workflow, the failed check cannot be due to commit.

I don't see a fail check

File: docs/reference/security.rst [hunk 1]
```diff
@@ -234,21 +234,23 @@ For example, you can `create your own voter`_
...
-            <argument type="collection">
-                <argument>ROLE_SUPER_ADMIN</argument>
-            </argument>
-        </service>
+        app.security.handler.role:
+            class="App\Security\Handler\RoleSecurityHandler"
+            public: false
+            arguments:
+                - '@security.authorization_checker'
+                - ['%sonata.admin.configuration.security.role_super_admin%']
```
Comment (VincentLanglet) [inline docs/reference/security.rst]: The declaration is not needed when autoconfigure/autowire is used.

Also, name should be less generic, like `App\Security\Handler\MySecurityHandler`.
And arguments depends on implementation.

I would remove this whole part and just say 

```markdown
If you want to change the handler behavior, create your own handler implementing 
``Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface``.

And specify it as Sonata security handler on your configuration:

sonata_admin:
            default_admin_services:
                security_handler: App\Security\Handler\MySecurityHandler
```

WDYT @AntoineRoue ?

Comment (AntoineRoue) [reply docs/reference/security.rst]: Yes I agree. I was hesitant about declaring this service. I did not check if the documentation assumes the user has services declared and autowired by default or not.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**